backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/allocation/transfer/HUTransformService.java

```diff
@@ -240,7 +241,14 @@ public Quantity getMaximumQtyCU(@NonNull final I_M_HU cu, @NonNull final I_C_UOM
 		final boolean qtyCuExceedsCuHU = qtyCU.compareTo(getMaximumQtyCU(cuOrAggregateHU, qtyCU.getUOM())) >= 0;
 		final boolean huIsCU = !handlingUnitsBL.isAggregateHU(cuOrAggregateHU);
 
-		if (qtyCuExceedsCuHU && huIsCU)
+		final IHUStorageFactory storageFactory = handlingUnitsBL.getStorageFactory();
+		final IHUStorage storage = storageFactory.getStorage(cuOrAggregateHU);
+
+		final I_C_UOM huUOM = storage.getC_UOMOrNull();
+		final UomId huUOMId = UomId.ofRepoId(huUOM == null ? -1 : huUOM.getC_UOM_ID());
+		final boolean isSameUOM = qtyCU.getUomId().equals(huUOMId);
+
+		if (qtyCuExceedsCuHU && huIsCU && isSameUOM)
```

Comment:
shouldn't we instead see if we can convert?

Reply:
If this condition is fulfilled, the source HU will be used as it is. Otherwise the HU is transformed and will fit the UOM of the qtyCU (parameter)  which is exactly what we want: to have all the HUs under the bom line in the same UOM as the bom line.